testrail/api.py

```diff
@@ -257,6 +263,29 @@ def run_with_id(self, run_id):
         except IndexError:
             raise TestRailError("Run ID '%s' was not found" % run_id)
 
+    @UpdateCache(_shared_state['_runs'])
+    def add_run(self, run):
+        fields = ['name', 'description', 'suite_id', 'milestone_id',
+                  'assignedto_id', 'include_all', 'case_ids']
+        project_id = run.get('project_id')
+        payload = self._payload_gen(fields, run)
+        return self._post('add_run/%s' % project_id, payload)
+
+    @UpdateCache(_shared_state['_runs'])
```

Comment:
I agree, and waffled on the proper approach when writing the decorator.

The response object for adding/updating/closing TestRail objects is the new/updated/closed object, so I think it should be possible to update the cache in place. Deletions return an empty dict, which should also make it possible to update the cache in place.

I will remove this functionality entirely from this PR, and add it to its own PR associated with issue #8.